src/Psalm/Internal/Type/SimpleNegatedAssertionReconciler.php

```diff
@@ -606,14 +609,17 @@ private static function reconcileNull(
                     $suppressed_issues,
                     $failed_reconciliation,
                     $is_equality
-                );
+                ));
 
+                // $did_remove_type = $did_remove_type || $new !== $type;
+                // TODO: This is technically wrong, but for some reason we get a
```

Comment:
is this something you want to work on still or is this for future reference?

Reply:
It's for future reference :)